bittensor/extrinsics/network.py

```diff
@@ -62,48 +69,58 @@ def register_subnetwork_extrinsic(
     wallet.coldkey  # unlock coldkey
 
     with bittensor.__console__.status(":satellite: Registering subnet..."):
-        with subtensor.substrate as substrate:
-            # create extrinsic call
-            call = substrate.compose_call(
-                call_module="SubtensorModule",
-                call_function="register_network",
-                call_params={"immunity_period": 0, "reg_allowed": True},
+        # create extrinsic call
+        call = await subtensor.substrate.compose_call(
+            call_module="SubtensorModule",
+            call_function="register_network",
+            call_params={"immunity_period": 0, "reg_allowed": True},
+        )
+        extrinsic = await subtensor.substrate.create_signed_extrinsic(
+            call=call, keypair=wallet.coldkey
+        )
+        response = await subtensor.substrate.submit_extrinsic(
+            extrinsic,
+            wait_for_inclusion=wait_for_inclusion,
+            wait_for_finalization=wait_for_finalization,
+        )
+
+        # We only wait here if we expect finalization.
+        if not wait_for_finalization and not wait_for_inclusion:
+            return True
+
+        # process if registration successful
+        # TODO: ? passably have to be awaited after `submit_extrinsic` implemented in `async_substrate.py`
```

Comment:
"Possibly". This is a good callout. Not yet finished with the extrinsics stuff in substrate.